client/index.js

```diff
@@ -0,0 +1,17 @@
+import {render} from "react-dom";
+import React from "react";
+
+import {DynamicList} from "./List";
```

Comment:
nit, but convention tends to be wrapping destructured values in parens.

`{DynamicList}` => `{ DynamicList }`

also

`{ DynamicList, Item }` if there's more than one.